### DISCUSSION TITLE
Asset URLs only working with a / at the end

### DISCUSSION THREAD

**Discussion by jimweeda** [snipe/snipe-it#14525] [Q&A] [answered]
After updating to the latest version (v6.3.3), I noticed that my assets only work if I add a / at the end.  If I just click the link of any asset, it gives me the 404 Page not found error.  As soon as I add the / at the end, it displayes the asset.  I'm not an expert so please consider that when anwsering or asking addational questions.  

Also, I found something that said to try to run the php upgrade.php again and it returns this error.

> ERROR: Failed to retrieve remote requirements from https://raw.githubusercontent.com/snipe/snipe-it/master/.upgrade_requirements.json
> 
> We suggest correcting this, but if you can't,  please verify that your requirements conform to those at that url.
> 
>  -- DANGER -- DO AT YOUR OWN RISK --
>       IF YOU ARE SURE, re-run this script with --skip-php-compatibility-checks to skip this check.
>  -- DANGER -- THIS COULD BREAK YOUR INSTALLATIONExiting.

Thank you in advance.

**Comment by snipe** [comment]
Latest is actually 6.3.4 :)

We didn't change anything in how URLs are parsed in recent versions, so I'm not sure why you'd be seeing that.

**Comment by snipe** [comment]
You can also upgrade manually - the upgrader just bundles up a bunch of commands that need to be run, so people don't accidentally miss a step.

```
git stash
git pull 
composer install --no-dev
php artisan migrate
php artisan config:clear
php artisan view:clear
php artisan cache:clear
```

Is there anything in your app logs that might indicate an issue somewhere else?

**Comment by jimweeda** [comment]
I was able to do the upgrade manually but had to add a step before composer install to upgrade composer.  After that, the upgrade worked but the URL issue still exists.  I did upgrade through several versions when I did the upgrades so was there a version somewhere back in the 4's or 5's that change URL parsing?

Thanks!

**Comment by jimweeda** [comment]
I wanted to provide an update.  After doing a bit more investigating, I realized that my URL actually had an extra / after the IP.  So I went into my .env file and removed that from behind the IP, saved it and it's working again.

**Comment by jimweeda** [comment, marked as answer]
My .env file had APP_URL=http://x.x.x.x/

I removed the trailing / so now it shows APP_URL=http://x.x.x.x

It is working again.
